src/components/TodoModal/TodoModal.tsx

```diff
@@ -2,11 +1,43 @@
+//TODOS ADD useEffect import, add props in args.
+
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { User } from '../../types/User';
+import { Todo } from '../../types/Todo';
+import { getUser } from '../../api';
+
+interface Props {
+  selectedTodo?: Todo;
+  setSelectedTodoId: (id: number | null) => void;
+}
+export const TodoModal: React.FC<Props> = props => {
+  const { selectedTodo, setSelectedTodoId } = props;
+  const [isLoading, setIsLoading] = useState(false);
+  const [user, setUser] = useState<User | null>(null);
+
+  useEffect(() => {
+    const getUserFromApi = async () => {
+      setIsLoading(true);
+      if (selectedTodo) {
+        try {
+          const userData = await getUser(selectedTodo.userId);
+
+          setUser(userData);
+          setIsLoading(false);
+        } catch (err) {
+          setIsLoading(true);
+          throw new Error('Error with getUserFromApi');
+        }
+      }
+    };
+
+    getUserFromApi();
+  });
```

Comment:
Dependency Array Issue: The `useEffect` dependency array is missing `selectedTodo`. Add `selectedTodo` to the dependency array to ensure the effect runs correctly when `selectedTodo` changes.